extensions/dash/src/hooks/useDashApp.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from "react";
+import { Application, getApplications } from "@raycast/api";
+import { usePersistentState } from "raycast-toolkit";
+
+export default function useDashApp(): [Application | null, boolean] {
+  const [dashApp, setDashApp, isLoading] = usePersistentState<Application | null>("dashApp", null);
+  useEffect(() => {
+    (async () => {
+      const applications = await getApplications();
+      for (const application of applications) {
+        if (application.bundleId?.startsWith("com.kapeli.dash")) {
+          setDashApp(application);
+          return;
+        }
+      }
+      throw new Error("Dash.app not found");
+    })();
+  }, []);
+
+  return [dashApp, isLoading];
+}
```

Comment:
I don't think it should be stored because it will lead to issues if the application changes.

I was just thinking about a local "cache":
```suggestion
let dashApp: Application | undefined

export default async function getDashApp(): Promise<Application> {
  if (dashApp) {
    return dashApp;
  }
  
  dashApp = (await getApplications()).find((app) => app.bundleId?.indexOf("com.kapeli.dash") === 0);

   if (!dashApp) {
     throw new Error("Dash.app not found");
   }

   return dashApp;
 }
```

Reply:
in that case it should be enough to just change `usePersistentState` to `useState` 😸